### ISSUE TITLE
Backdating in 1.94.0 should show supplied, warn about posting time

### ISSUE THREAD

**Issue by wlonkly** [bluesky-social/social-app#6707]
### Describe the Feature

(I couldn't decide if this should be a feature request or a bug report, going with feature request for now. I've avoided using the terms `createdAt` and `indexedAt` avoid confusion -- there's a date _supplied_ with the skeet, and there's the date the skeet was _received_.)

As of 1.94.0, imported skeets show a warning that they're imported, which is a welcome change because being able to _quietly_ backdate a skeet is ripe for abuse.

But I am going to suggest that the **dates are backwards**.  

I was surprised there was no discussion of it in https://github.com/bluesky-social/social-app/pull/6156 (and I couldn't find a related Github issue). 

As I see it, we're trying to express "The supplied date on this skeet is not trustworthy because we received it much later than the supplied date."

**What I would expect** from a backdated skeet is for the supplied date to be used for display and for chronological sorting, and then a server-side warning about when the skeet was actually received:

<img width="500" src="https://github.com/user-attachments/assets/aa335773-aa8a-43b4-bdc3-81159a5b11a4">

(also note wording change "posted on"; the verb "archived" doesn't make much sense to me here, we don't know _why_ the dates differ, and the date I archived my twitter posts was months before I imported them to Bluesky)

**What we have now** is the other way on: a backdated skeet appears with its received date, and the supplied date is only available visible inside the warning: 

<img width="500" src="https://github.com/user-attachments/assets/ca2f789d-10f8-4b80-931f-a46b631b6633">

Since the main purpose of backdating right now is importing old content -- and since people are being surprised because their Twitter etc. imports are now showing up as the most recent posts in their feed -- I'd encourage switching this around. 

As an aside, the text inside the popup is clear (other than "Archived", but that's only a secondary issue here). 

<img width="362" alt="image" src="https://github.com/user-attachments/assets/3ed75d78-e49a-4878-aa42-79d9e8af7eef">

**Comment by brw** [comment]
I did see Samuel saying this is intentional here https://bsky.app/profile/samuel.bsky.team/post/3lbnczv6hvs2t

I could also see the reverse being better considering backdating is technically a feature, so perhaps it should be regarded as such. The warning label should make it obvious enough to know that the date has likely been tampered with. Maybe it should show in the feed as well.

**Comment by wlonkly** [comment]
Ah, thanks for the pointer.  Perhaps one happy medium is to have the supplied date shown as "the date", but with the calendar icon (or a calendar icon with a ⚠️ incorporated) _there_, and then the label inside the post. 

Not my proudest mockup, but you get the idea: 

<img width="607" alt="image" src="https://github.com/user-attachments/assets/8b2efd13-545c-4318-9df2-84197fe6b985">

**Comment by FernsDC** [comment]
> **What I would expect** from a backdated skeet is for the supplied date to be used for display and for chronological sorting, and then a server-side warning about when the skeet was actually received

As someone importing old tweets, I 100% agree and support this.

Not only because of the reasons already stated, but because the way this has been implemented causes confusion both in time line chronology & in searches, which both use the 'supplied date' (i.e. the original timestamp) despite that not being the date that's displayed to the user.

TL:
Scrolling back in my TL, I suddenly run into posts from '3d ago' wayyyy down the list because they're actually sorted (correctly) by the supplied date, but they display the received date in the TL.

SEARCH:
Similarly, when I search for a date range that includes older imported posts, the results are correct per the supplied date, but the search results show posts that are labelled as having been created a few days ago. 

Both of these make for a very confusing user experience because it's illogical, and it seriously detracts from the value and of importing old posts which is to keep *history* and for it to be seen as such.

**Comment by jamesleesaunders** [comment]
Hi, I did not realise this had already been raised, but I too was a little put out when the backdated posts indicator feature was implemented.

I have raised a pull request which I think goes some way to address this https://github.com/bluesky-social/social-app/pull/7466

**Comment by cooljeanius** [comment]
Yeah this led to some confusion on one of my BlueArk-imported posts: https://bsky.app/profile/cooljeaniusarchive.bsky.social/post/3lhteush3na24

**Comment by davidcelis** [comment]
+1, if the post is going to get a warning message on it, I feel like the social app may as well still display the date as the post's `createdAt` timestamp rather than the "first seen at" timestamp.

It also seems like the badge is applied inconsistently. For example, this post was backdated with a `createdAt` about 15-16 hours earlier, but there's no badge: https://bsky.app/profile/davidcel.is/post/3lovwxuu5w42w

**Comment by jamesleesaunders** [comment]
> It also seems like the badge is applied inconsistently. For example, this post was backdated with a `createdAt` about 15-16 hours earlier, but there's no badge: https://bsky.app/profile/davidcel.is/post/3lovwxuu5w42w

I can answer that one... The current backdate warning only shows if if it over 24 hours difference between the indexedAt and createdAt dates...

```
  // backdated if createdAt is 24 hours or more before indexedAt
  const isBackdated =
    indexedAt.getTime() - createdAt.getTime() > 24 * 60 * 60 * 1000
```

So 15-16 hours will not trigger it.

**Comment by cooljeanius** [comment]
I had been using a labeller to work around this, but apparently it's no longer running... see discussion in mozzius/backdate-labeller#1

**Comment by hexadecimal233** [comment]
It is so frustrated that the issue is still open in 2026

**Comment by cooljeanius** [comment]
> It is so frustrated that the issue is still open in 2026

It really wouldn't be that much of a problem if @mozzius had just kept the backdate-labeller running, but alas...
